PathPlanning/DStar/dstar.py

```diff
@@ -189,17 +198,31 @@ def modify(self, state):
 
 def main():
     m = Map(20, 20)
-    obstacle_list = [(8, 3), (8, 4), (8, 5), (8, 6), (8, 7),
-                     (8, 8), (8, 9), (8, 10), (8, 11), (8, 12),
-                     (8, 13), (7, 13), (6, 13), (5, 13)]
-
-    m.set_obstacle(obstacle_list)
+    ox, oy = [], []
+    for i in range(-10, 60):
+        ox.append(i)
+        oy.append(-10.0)
+    for i in range(-10, 60):
+        ox.append(60.0)
+        oy.append(i)
+    for i in range(-10, 61):
+        ox.append(i)
+        oy.append(60.0)
+    for i in range(-10, 61):
+        ox.append(-10.0)
+        oy.append(i)
+    for i in range(-10, 40):
+        ox.append(20.0)
+        oy.append(i)
+    for i in range(0, 40):
+        ox.append(40.0)
+        oy.append(60.0 - i)
+    m.set_obstacle([[i, j] for i, j in zip(ox, oy)])
 
     start = [1, 2]
     goal = [17, 19]
     if show_animation:
-        plt.plot([x for (x, y) in obstacle_list],
-                 [y for (x, y) in obstacle_list], ".k")
+        plt.plot([ox, oy, ".k"])
```

Comment:
I think this should be `plt.plot(ox, oy, ".k")`? In my environment, simulation has an error: 

```
/usr/local/lib/python3.9/site-packages/numpy/core/_asarray.py:171: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  return array(a, dtype, copy=False, order=order, subok=True)
TypeError: float() argument must be a string or a number, not 'list'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/atsushisakai/PycharmProjects/PythonRobotics/PathPlanning/DStar/dstar.py", line 240, in <module>
    main()
  File "/Users/atsushisakai/PycharmProjects/PythonRobotics/PathPlanning/DStar/dstar.py", line 225, in main
    plt.plot([ox, oy, ".k"])
  File "/usr/local/lib/python3.9/site-packages/matplotlib/pyplot.py", line 2988, in plot
    return gca().plot(
  File "/usr/local/lib/python3.9/site-packages/matplotlib/axes/_axes.py", line 1607, in plot
    self.add_line(line)
  File "/usr/local/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 2105, in add_line
    self._update_line_limits(line)
  File "/usr/local/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 2127, in _update_line_limits
    path = line.get_path()
  File "/usr/local/lib/python3.9/site-packages/matplotlib/lines.py", line 1022, in get_path
    self.recache()
  File "/usr/local/lib/python3.9/site-packages/matplotlib/lines.py", line 668, in recache
    y = _to_unmasked_float_array(yconv).ravel()
  File "/usr/local/lib/python3.9/site-packages/matplotlib/cbook/__init__.py", line 1333, in _to_unmasked_float_array
    return np.asarray(x, float)
  File "/usr/local/lib/python3.9/site-packages/numpy/core/_asarray.py", line 102, in asarray
    return array(a, dtype, copy=False, order=order)
ValueError: setting an array element with a sequence.
```

Reply:
you are right...I have fixed it. Sorry for delay.